src/lib.rs

```diff
@@ -2216,7 +2217,13 @@ pub fn process_frame(sequence: &mut Sequence, fi: &mut FrameInvariants,
                 let rec_frame = y4m::Frame::new([&rec_y, &rec_u, &rec_v], None);
                 y4m_enc.write_frame(&rec_frame).unwrap();
             }
-            *last_rec = Some(fs.rec);
+
+            let rfs = Rc::new(fs.rec);
```

Comment:
The test below probably needs the same code (and maybe it could be wrapped in a tiny inline function)